File: algorithms/JavaScript/XorSwap/XorSwap.js
/**
 * xorSwap
 *
 * Swaps two variables without using a temporary variable
 *
 */
function xorSwap(a, b) {
  let tempA = a;
  let tempB = b;
  tempA ^= tempB;
  tempB = tempA ^ tempA;
  tempA ^= tempB;

  return { a: tempA, b: tempB };
}

module.exports = { xorSwap };

File: algorithms/bit-manipulation/xor-swap/go/XorSwap.go
package xorswap

// XorSwap swaps two integers using XOR without a temporary variable.
func XorSwap(a, b int) (int, int) {
	if a != b {
		a = a ^ b
		b = a ^ b
		a = a ^ b
	}
	return a, b
}
